File: an2_sem4/MPP/Homework/sapt14/service/src/main/java/ro/ubbcluj/cs/stefnmubb/festivalsellpoint/service/SpectacolService.java
package ro.ubbcluj.cs.stefnmubb.festivalsellpoint.service;

import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.domain.Spectacol;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.repo.EntityRepoException;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.repo.ISpectacolRepo;

import java.time.LocalDateTime;

public class SpectacolService extends AbstractService<Integer, Spectacol, ISpectacolRepo>
    implements ISpectacolService{
    public SpectacolService(ISpectacolRepo repo) {
        super(repo);
    }

    @Override
    public Iterable<Spectacol> getBetweenDates(LocalDateTime start, LocalDateTime end) throws EntityRepoException {
        return getRepo().getBetweenDates(start, end);
    }
}

File: an2_sem4/MPP/Homework/sapt10/cs/FestivalSellpoint/FestivalSellpoint.Service/SpectacolService.cs
using FestivalSellpoint.Domain;
using FestivalSellpoint.Repo;
using System;
using System.Collections.Generic;

namespace FestivalSellpoint.Service
{
    public class SpectacolService : AbstractService<int, Spectacol, ISpectacolRepo>, ISpectacolService
    {
        public SpectacolService(ISpectacolRepo repo) : base(repo)
        {
        }

        public IEnumerable<Spectacol> GetBetweenDates(DateTime start, DateTime end)
            => Repo.GetBetweenDates(start, end);
    }
}
